Added Auto detect + minor changes

kevinf100 opened the pull request:
Added Auto detect buttons for Steam and GoG.
Code uses windows liberty's to read the registry for the locations.
It has some basic error checking, needs improvement. 
Disable resizing.

I recommend you merge https://github.com/LittleZen/Cyberpunk2077-Patch-Easy-Installer/pull/4 and I'll update this based around that since he does a good job cleaning up the code.

kevinf100 committed:
Added Auto detect + minor changes

Added Auto detect buttons for Steam and GoG.
Code uses windows liberty's to read the registry for the locations.
It has some basic error checking, needs improvement. 
Disable resizing.
Moved the code to move the patch to a function that takes in a string and uses that as a path.

LittleZen commented:
Hey Kevin, thank you for your review! 

I'm currently work on #4 , once finish i will check this merge and work on it too!

Thank you for your interesting, i will keep you informed !

LittleZen commented:
@kevinf100. I merged (successfully) #4 , but seems it causing some conflict with your currently PR. I also made some other small changes, can you check it ?

kevinf100 committed:
Merge pull request #2 from LittleZen/master

Update  to latest

Commit:
Added Auto detect buttons for Steam and GoG.
Code uses windows liberty's to read the registry for the locations.
It has some basic error checking, needs improvement.
Disable resizing.
Moved the code to move the patch to a function that takes in a string and uses that as a path.
FIxed tab index

kevinf100 committed:
Merge branch 'kevinf100-patch-Auto-Detect' into AutoUpdate-Stand

kevinf100 committed:
Merge pull request #3 from kevinf100/AutoUpdate-Stand

Auto update stand

Commit:
Updated Message box to Metro Style.
Cancel is not longer a crystal error.

kevinf100 commented:
Done.
Updated for current Master.
Also made cancel no longer a critical error as it is not.
Also note that I threw them in a try-catch that will catch all exceptions. Any that arise I will update the try-catch and/or FindGames.cs if needed.
Fun fact FindGames.cs will work for any Steam or GoG game.

LittleZen commented:
@kevinf100  Hey man, thank you for your hard work. 

I've currently released the version 2.2, which is another time in conflict with your merge. I'm so sorry for this inconvenience, i will stop to work on new release for a wile for let you the opportunity to correctly merge all of this interesting features!

Another time, i'm so sorry for the conflicts, hope you can release your hard work !

kevinf100 committed:
Merge pull request #4 from LittleZen/master

Update

Commit:
Update

Commit:
Update

LittleZen commented:
@kevinf100  Have you already test it with version 2.2 ?

kevinf100 committed:
Merge pull request #6 from LittleZen/master

Update

Commit:
Update tab index

Commit:
Merge branch 'master' into kevinf100-patch-Auto-Detect

kevinf100 commented:
Should be all set hopefully.
A note that it only auto finds when the user presses one of the buttons. If for whatever reason the user crashes when they hit one of them they will not stay in a loop of crashing.

LittleZen commented:
Perfect ! I've released the version 2.2 now, so I'll take a break for a while (understand me, i'm dying hahaha), and then i will check your merge.

Another time, thank you for your hard work, i will keep you informed <3

kevinf100 committed:
Merge pull request #7 from LittleZen/master

Update

LittleZen commented:
Hey @kevinf100 !

i'm sorry, but i make a mess with the version 2.2, and i was forced to quickly released the version 2.3 for fix it. I swear, is the last time, i was looking at your code, meanwhile other guys opened issues and i was forced to quickly reply!


If you could please fix the conflict with the new patch 2.3 i will really appreciate that, and i will instantly merge everything !


So sorry for that :(

Commit:
Updated to latest changes

kevinf100 commented:
Conflicts fixed. As well as tab indexes updated.

Commit:
Updated Button names

LittleZen commented:
Amazing work <3 released and attached with version 2.3

kevinf100 commented:
https://github.com/kevinf100/Find-Steam-or-GoG-game-by-appID/blob/main/FindGames.cs
for GoG all game installs are saved in "HKEY_LOCAL_MACHINE\\SOFTWARE\\WOW6432Node\\GOG.com\\Games\\" with the GameID as a folder(? I think its called). The GameID folder contains a "PATH" key with the game path as a value. Nothing special. 
Steam is a bit harder. Your steam install is in "HKEY_LOCAL_MACHINE\\SOFTWARE\\Wow6432Node\\Valve\\Steam" with key "InstallPath". Where ever steam is installed is already a game library. Next steam stores all game library's (excluding steam install directory) in "(steam path)\\steamapps\\libraryfolders.vdf". Using all this we get a list of all locations a game can be for steam. Next every steam game has a .acf file. So we search each steam library "steamapps" until we find one with our appID. The .acf file also holds the folder name for the game. So using all this info we can build the path
So yes SteamGamePath.FindGameByAppID(730) will return the path to the main game folder and so will GoGGamePath.FindGameByAppID(730)
If it can't find it, it will return a null instead. These can still throw filestream errors if it encounters problems tho.

LittleZen commented:
@kevinf100  Thank you man, i will check all the documentation as soon as possible. 

The problem is: the "auto-find" system, correctly works, but it didn't return a valid path for the others settings ( check "save settings", "Open Patch Logs", "Open Json Settings" and "check for updates" ) even if the "auto detect" system seems to works correctly these options rest unusable

LittleZen commented:
I'm fixing it! Thank you, i will check the documentation passed.

JayVee732 commented on CP2077 - EasyInstall/Form1.cs:
```diff
@@ -366,5 +378,63 @@ private void btnUninstall_Click(object sender, EventArgs e)
+        {
+            try
+            {
+                string path = GoGGamePath.FindGameByAppID("1423049311");
```
Hi all,

I've tested out the auto find with the GoG version of the game and while it returns the root Cyberpunk installation, it's not the correct path for the files to be placed in.

We either need to return the path with `\bin\x64` already appended, or append it to the `PatchGame(path)` method below (See below comment).

JayVee732 commented on CP2077 - EasyInstall/Form1.cs:
```diff
@@ -366,5 +378,63 @@ private void btnUninstall_Click(object sender, EventArgs e)
+                //DialogResult dialogResult = MessageBox.Show(path, "Is this Correct?", MessageBoxButtons.YesNo);
+                DialogResult result = MetroFramework.MetroMessageBox.Show(this, path, "Is this Correct?", MessageBoxButtons.YesNo);
+                if (result == DialogResult.Yes)
+                    PatchGame(path);
```
```suggestion
                    PatchGame($"{path}\\bin\\x64");
```
This will work, but it depends on where you want to append the additional parts of the path.

kevinf100 replied:
Ahh my bad. I assumed PatchGame added "bin\x64" to it. Not that you had to pass the x64 path

LittleZen commented:
Hey @JayVee732! Can you make a new PR for correctly merge it ?
